deployment-server/src/Deploy/Worker.hs

```diff
@@ -103,18 +109,35 @@ runIn dir bin args = do
             liftIO $ print stderr
             throwError $ CommandError ("Failed with exit code " <> (Text.pack . show) n <> " for command: " <> (Text.pack . unwords) ([dir, bin] <> args))
     where
-        otherErrors :: SomeException -> IO (ExitCode, BS.ByteString, BS.ByteString)
-        otherErrors e = pure (ExitFailure 1, mempty, BS.pack (displayException e))
+        otherErrors :: SomeException -> IO (ExitCode, LBS.ByteString, LBS.ByteString)
+        otherErrors e = pure (ExitFailure 1, mempty, LBS.pack (displayException e))
 
-showResult :: Either DeploymentError Text -> Text
-showResult (Left (CommandError err)) = "failed to deploy with error: " <> err
-showResult (Right gitHead)           = "git ref " <> gitHead <> " deployed successfully"
+createGithubDeployment :: String -> String -> GitHub.Auth -> IO (Either GitHub.Error (GitHub.Deployment String))
+createGithubDeployment ref environment auth = do
+    let deployment = GitHub.CreateDeployment
+            { createDeploymentRef = Text.pack ref
+            , createDeploymentTask = Nothing
+            , createDeploymentAutoMerge = Just False
+            , createDeploymentRequiredContexts = Nothing
+            -- despite being `Maybe a` this fails if the value is `Nothing`
+            , createDeploymentPayload = Just ""
+            , createDeploymentEnvironment = Just $ Text.pack environment
+            , createDeploymentDescription = Just "Deploy Playgrounds"
+            }
+        req = GitHub.createDeploymentR "input-output-hk" "plutus" deployment
+    GitHub.executeRequest auth req
 
-alert :: SlackConfig -> SlackChannel -> Text -> IO ()
-alert config channel message = do
-    let msg =
-            mkPostMsgReq (unpack channel) message
-    result <- flip runReaderT config $ chatPostMessage msg
-    case result of
-        Right _ -> pure ()
-        Left e  -> putStrLn $ "Failed to notify slack with error " <> show e
+completeGithubDeployment :: GitHub.Auth -> GitHub.Deployment String -> Either DeploymentError b -> IO (Either GitHub.Error GitHub.DeploymentStatus)
+completeGithubDeployment auth deployment result = do
+    let dId = GitHub.deploymentId deployment
+        deploymentStatus = case result of
+                            Right _ -> GitHub.CreateDeploymentStatus
+                                            GitHub.DeploymentStatusSuccess
+                                            Nothing
+                                            (Just "Deployed Successfully")
+                            Left err -> GitHub.CreateDeploymentStatus
+                                            GitHub.DeploymentStatusFailure
+                                            Nothing
+                                            (Just . Text.pack $ "Deployment failed with " <> take 117 (show err))
```

Comment:
What's this magic number about? :-D

Reply:
added a comment - the description is limited to 140 characters